### PR TITLE
Use MAC addresses to create vSwitch uplinks and networkDevices in esx-ks

### PR DESCRIPTION
**Pull request by lanchongyizu**
1. Fix ODR-749 Use MAC addresses to create vSwitch uplinks in esx-ks
2. Use MAC addresses to specify the device under networkDevices

@RackHD/corecommitters @panpan0000

### CODE REVIEW

**Comment by lanchongyizu** [comment]
@zyoung51 Please help to review this PR. Thanks.

**Comment by panpan0000** [comment]
@yyscamper @anhou , any chance to approve it ? alone with https://github.com/RackHD/on-tasks/pull/271

File: data/templates/esx-ks [hunk 1]
```diff
@@ -8,24 +8,22 @@ accepteula
...
+      <% if (typeof n.ipv4.vlanIds !== 'undefined' ) { %>
+        <% ipopts += ' --vlandid=' + n.ipv4.vlanIds[0] %>
       <% } %>
-      <% return false; %>
+      network --bootproto=static --device=<%=n.device%> <%=ipopts%>
+    <% } else { %>
+      network --bootproto=dhcp --device=<%=n.device%>
     <% } %>
-    <% return true; %>
+    <% return false; %>
```
Comment (zyoung51) [inline data/templates/esx-ks]: In the original template the 'device' name maps to the portgroup name and does not have to be the physical interface used to boot.  Returning 'false' here breaks that and forces the first interface to be the physical interface that is being used to load ESXi.  This new behavior should probably be documented somewhere.

**Comment by zyoung51** [comment]
+1

@lanchongyizu added an ask for some comment in the template or documentation in readthedocs about the requirement that the first networkDevice be the boot network device.

**Comment by lanchongyizu** [comment]
@zyoung51 OK, I will make changes to the comment in the template and documentation.
Thanks.

**Comment by yyscamper** [comment]
:+1:

**Pull request merged**